onnxruntime/core/session/inference_session.h

```diff
@@ -627,8 +627,8 @@ class InferenceSession {
   template <typename T>
   void StartProfiling(const std::basic_string<T>& file_prefix);
 
-  // Updates all providers with the allocators from the env based on OrtMemoryInfo
-  void UpdateProvidersWithSharedAllocators();
+  // Updates all SessionState allocators with the allocators from the env
+  void UpdateSessionStateAllocatorsWithSharedAllocators();
```

Comment:
Shared is super confusing here. May be this should be called UpdateSessionStateWithExternalAllocators or UpdateSessionStateWithEnvAllocators?

Reply:
Sure I will change the name in next iteration